docs/advanced/automated-testing.md

```diff
@@ -0,0 +1,262 @@
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js [16.x]
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.x
+          cache: "npm"
+      - name: Npm Install && Build with Testing preset
+        run: npm install && npm run-script build
+      - name: Install Lua
+        uses: leafo/gh-actions-lua@v8
+        with:
+          luaVersion: "luajit-2.1.0-beta3"
+      - name: Install LuaRocks
+        uses: leafo/gh-actions-luarocks@v4.0.0
+      - name: Install Busted
+        run: luarocks install busted
+      - name: Running Tests
+        run: cd dist && busted
+```
+
+Without going in to much detail, this workflow triggers whenever you push a commit to your github repository, installs Node.js, Lua, Luarocks and Busted installs all of your project dependencies and compiles a fresh version of the project, at the very end it moves in to the dist folder and runs your busted tests exactly like you would. Now, if you commit and push these changes to your GitHub repository you should be able to see, Under the Actions tab at the top, something like this:
+
+![Github action result](/images/github-workflow-result.png)
+
+Now that’s just it, this job will run every time you push a new commit to the repository, if it fails it’ll show up as a red cross, and you can find which test failed in the logs.
+
+## TSTL Library specifics
+
+This section applies to projects described in [Publishing Modules](publishing-modules.md), as you might know, library projects don’t (and shouldn’t) include the luaLibBundle and also any dependencies it would need to run, so how do we make it run standalone, for testing, we’re going to utilize the fact that you can pass in a different `tsconfig.json` to the compiler, so, the regular one is implicitly used based on the name `tsconfig.json`, we’re going to create a new one like `tsconfig.test.json` you can pretty much duplicate it from your current one, we’re just going to change a few values:
+
+```json title=tsconfig.test.json
+{
+  "compilerOptions": {
+    "lib": ["esnext"],
+    "rootDir": "src",
+    "outDir": "dist" => "testing",
+    "target": "esnext",
+    "moduleResolution": "node",
+    "types": [
+      "typescript-to-lua/language-extensions",
+      "lua-types/jit",
+      "busted-tstl"
+    ],
+    "declaration": true,
+    "noImplicitAny": true,
+  },
+  "tstl": {
+    "luaLibImport": "none" => "require",
+    "buildMode": "library" => "default",
+    "luaTarget": "JIT",
+    "noImplicitSelf": true,
+    "sourceMapTraceback": false
+  }
+}
+```
+
+_(arrows are to indicate the changes, don't include them in the actual code)_
+
+Now you can use this new tsconfig like: `tstl —-project tsconfig.test.json`
+
+It will output the new build in to the testing folder and you can move your terminal in to the folder and run busted
+
+Then you can edit your `package.json` file and add a new task:
+
+```json title=package.json
```

Comment:
using `diff` instead of `json` highlighting might be better for this one:
```diff title=package.json
  "scripts": {
    "build": "tstl",
+   "test-build": "tstl --project tsconfig.test.json",
    "dev": "tstl --watch"
  },
```